File: src/main/scala/mrtjp/projectred/expansion/item/EmptyBatteryItem.scala
package mrtjp.projectred.expansion.item

import mrtjp.projectred.expansion.ExpansionContent
import net.minecraft.item.Item

class EmptyBatteryItem extends Item(new Item.Properties().group(ExpansionContent.expansionItemGroup)) with TChargableBatteryItem
{
    override def isEmpty = true

    override def getEmptyVariant:Item = this

    override def getChargedVariant:Item = ExpansionContent.batteryItem.get()
}

File: src/expansion/java/mrtjp/projectred/expansion/item/EmptyBatteryItem.java
package mrtjp.projectred.expansion.item;

import mrtjp.projectred.ProjectRedExpansion;
import mrtjp.projectred.expansion.init.ExpansionReferences;
import net.minecraft.item.Item;

public class EmptyBatteryItem extends Item implements IRechargableBattery {

    public EmptyBatteryItem() {
        super(new Item.Properties().tab(ProjectRedExpansion.EXPANSION_GROUP));
    }

    @Override
    public Item getChargedVariant() {
        return ExpansionReferences.BATTERY_ITEM;
    }

    @Override
    public Item getEmptyVariant() {
        return this;
    }
}
